tools/fastlane-plugin/lib/fastlane/plugin/bugsnag/actions/send_build_to_bugsnag.rb

```diff
@@ -172,72 +172,96 @@ def self.available_options
+        rescue
+        end
+        return git_options
+      end
+
+      # Used to get a default configuration file (AndroidManifest.xml or Info.plist)
+      def self.default_config_file
         case lane_context[:PLATFORM_NAME]
         when nil
           if file_path = default_android_manifest_path
-            options.merge!(load_default_android_values(file_path))
+            return file_path
           elsif file_path = default_info_plist_path
-            options.merge!(options_from_info_plist(file_path))
+            return file_path
           end
         when :android
           if file_path = default_android_manifest_path
-            options.merge!(load_default_android_values(file_path))
+            return file_path
           end
         else
           if file_path = default_info_plist_path
-            options.merge!(options_from_info_plist(file_path))
+            return file_path
           end
         end
-
-        if git_opts = git_remote_options
-          options.merge!(git_opts)
-        end
-        options
-      end
-
-      def self.load_default_android_values file_path
-        options = options_from_android_manifest(file_path)
-        build_gradle_path = Dir.glob("{android/,}app/build.gradle").first
-        build_gradle_path ||= Dir.glob("build.gradle").first
-        options.merge!(options_from_build_gradle(build_gradle_path)) if build_gradle_path
-        options
       end
 
-      def self.default_android_manifest_path
-        Dir.glob("./{android/,}{app,}/src/main/AndroidManifest.xml").first
+      def self.load_config_file_options config_file
+        options = {}
+        case File.extname(config_file)
+        when ".xml"
+          options = load_options_from_xml(config_file)
+          return options
+        when ".plist"
+          options = load_options_from_plist(config_file)
+          return options
+        else
+          UI.user_error("File type of '#{config_file}' was not recognised. This should be .xml for Android and .plist for Cococa")
+        end
       end
 
       def self.default_info_plist_path
         Dir.glob("./{ios/,}*/Info.plist").reject {|path| path =~ /build|test/i }.first
       end
-
-      def self.options_from_info_plist file_path
+      def self.load_options_from_plist file_path
+        options = {}
         plist_getter = Fastlane::Actions::GetInfoPlistValueAction
-        {
-          apiKey: plist_getter.run(path: file_path, key: "BugsnagAPIKey"),
-          appVersion: plist_getter.run(path: file_path, key: "CFBundleShortVersionString"),
-          appBundleVersion: plist_getter.run(path: file_path, key: "CFBundleVersion"),
-          config_file: file_path,
-        }
+        bugsnag_dict = plist_getter.run(path: file_path, key: "bugsnag")
+        api_key = bugsnag_dict["apiKey"] unless bugsnag_dict.nil?
+        if api_key.nil?
+          api_key = plist_getter.run(path: file_path, key: "BugsnagAPIKey")
+        end
+        options[:apiKey] = api_key
```

Comment:
Don't we support releaseStage in the plist?

Reply:
I don't believe so. None of the default plists show a release stage AFAIK.

Reply:
@tomlongridge ? I thought everything was going to be configurable in the plist now?

Reply:
I can add it in. Suggest maybe putting it alongside the apiKey so it's `bugsnag.releaseStage`?

Reply:
Yes - all configuration options (that are expressable) can now be read from the plist in the notifier. I hadn't extended it this far, but it makes perfect sense to read this.

@xander-jones - the options and their locations are already set and can be found in the [docs](https://docs.bugsnag.com/platforms/ios/configuration-options/#releasestage). But, yes, `bugsnag.releaseStage` is the correct location.

Reply:
Done: https://github.com/bugsnag/bugsnag-dsym-upload/pull/42/commits/67fb72dab7ceba389f4bec58b491eec6f02728f6 Falls back to `production` if no params provided, or not found in the plist.